sycl/test-e2e/ESIMD/unified_memory_api/Inputs/atomic_update.hpp

```diff
@@ -262,103 +274,114 @@ struct ImplInc : ImplIncBase<T, N, atomic_op, atomic_op::inc> {};
 bool test_int_types(queue q, const Config &cfg) {
   bool passed = true;
   if constexpr (SignMask & Signed) {
     // Supported by LSC atomic:
-    if constexpr (UseProperties)
-      passed &= test<int16_t, N, Op, UseProperties>(q, cfg);
+    if constexpr (UsePVCFeatures)
+      passed &= test<int16_t, N, Op, UseMask, UsePVCFeatures>(q, cfg);
 
-    passed &= test<int32_t, N, Op, UseProperties>(q, cfg);
+    passed &= test<int32_t, N, Op, UseMask, UsePVCFeatures>(q, cfg);
   }
 
   if constexpr (SignMask & Unsigned) {
     // Supported by LSC atomic:
-    if constexpr (UseProperties)
-      passed &= test<uint16_t, N, Op, UseProperties>(q, cfg);
+    if constexpr (UsePVCFeatures)
+      passed &= test<uint16_t, N, Op, UseMask, UsePVCFeatures>(q, cfg);
 
-    passed &= test<uint32_t, N, Op, UseProperties>(q, cfg);
+    passed &= test<uint32_t, N, Op, UseMask, UsePVCFeatures>(q, cfg);
   }
   return passed;
 }
 
-template <int N, template <class, int> class Op, bool UseProperties>
+template <int N, template <class, int> class Op, bool UseMask,
+          bool UsePVCFeatures>
 bool test_fp_types(queue q, const Config &cfg) {
   bool passed = true;
-  passed &= test<float, N, Op, UseProperties>(q, cfg);
+  passed &= test<float, N, Op, UseMask, UsePVCFeatures>(q, cfg);
   if (q.get_device().has(sycl::aspect::atomic64) &&
       q.get_device().has(sycl::aspect::fp64)) {
-    // Disable double data type for fcmpxchg operation as D64 data is not
-    // supported for that operation.
-    passed &= test<double, N, Op, UseProperties>(q, cfg);
+    passed &= test<double, N, Op, UseMask, UsePVCFeatures>(q, cfg);
   }
   return passed;
 }
 
-template <template <class, int> class Op, bool UseProperties,
+template <template <class, int> class Op, bool UseMask, bool UsePVCFeatures,
           int SignMask = (Signed | Unsigned)>
 bool test_int_types_and_sizes(queue q, const Config &cfg) {
   bool passed = true;
 
-  passed &= test_int_types<1, Op, UseProperties, SignMask>(q, cfg);
-  passed &= test_int_types<2, Op, UseProperties, SignMask>(q, cfg);
-  passed &= test_int_types<4, Op, UseProperties, SignMask>(q, cfg);
+  passed &= test_int_types<1, Op, UseMask, UsePVCFeatures, SignMask>(q, cfg);
+  passed &= test_int_types<2, Op, UseMask, UsePVCFeatures, SignMask>(q, cfg);
+  passed &= test_int_types<4, Op, UseMask, UsePVCFeatures, SignMask>(q, cfg);
 
-  passed &= test_int_types<8, Op, UseProperties, SignMask>(q, cfg);
+  passed &= test_int_types<8, Op, UseMask, UsePVCFeatures, SignMask>(q, cfg);
 
   // Supported by LSC atomic:
-  if constexpr (UseProperties) {
-    passed &= test_int_types<16, Op, UseProperties, SignMask>(q, cfg);
-    passed &= test_int_types<32, Op, UseProperties, SignMask>(q, cfg);
+  if constexpr (UsePVCFeatures) {
+    passed &= test_int_types<16, Op, UseMask, UsePVCFeatures, SignMask>(q, cfg);
+    passed &= test_int_types<32, Op, UseMask, UsePVCFeatures, SignMask>(q, cfg);
```

Comment:
The implementation calls LSC atomic_update() if N is not powerOfTwo(32).
Please add couple of cases, e.g. N=12, N=33;

Reply:
+ same for FP types.

Reply:
Done in https://github.com/intel/llvm/pull/11833/commits/4689604e36d22087fb0a63da4837be828cca8888.